src/board.js

```diff
@@ -0,0 +1,154 @@
+// This is the board as an array
+let board = [
+  [0, 0, 0],
+  [0, 0, 0],
+  [0, 0, 0],
+];
+
+const isDraw = () => {
+  const allDifferent = board.every(row => row.every(col => col !== 0));
+
+  if (allDifferent) return 'draw';
+
+  return 'go';
+};
+
+// This is for checking win or draw
+const status = () => {
+  // these two variables keep track of the winner
+  let winner = false;
+  let value;
+  let player;
+  let row;
+
+  const slotsAreEqual = (row, value) => row.every(slot => slot === value);
+
+  // in here we iterate over the rows to check for a winner
+  for (let i = 0; i < 3; i += 1) {
+    row = board[i];
+    [value] = row;
+
+    if (value !== 0) {
+      winner = slotsAreEqual(row, value);
+
+      player = value === 1 ? 'x' : 'o';
+
+      if (winner) return player;
+    }
+  }
+
+
+  // in here we check for a diagonal with all the values equal
+  const leftToRightDiagonal = [];
+
+  for (let i = 0; i < board.length; i += 1) {
+    if (board[i][i] === 0) break;
+    leftToRightDiagonal.push(board[i][i]);
+  }
+
+  if (leftToRightDiagonal.length === 3) {
+    winner = leftToRightDiagonal.every(slot => slot === leftToRightDiagonal[0]);
+
+    [value] = leftToRightDiagonal;
+    player = value === 1 ? 'x' : 'o';
+
+    if (winner) return player;
+  }
+
+  // in this part we go for the columns
+  const columns = []; let
+    column;
+
+  for (let i = 0; i < 3; i += 1) {
+    column = [];
+
+    for (let j = 0; j < 3; j += 1) {
+      if (board[j][i] === 0) break;
```

Comment:
Your check for a winner can be greatly improved ,  instead of checking for a column, diagonal, row seperately you can create a moves array ![see this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png) that stores all players move then and a winning combination that stores all possible winning combo, then check the moves against the winning combination if it exist if yes you have a winner :slightly_smiling_face:

Reply:
Good idea!